### ISSUE TITLE
Parser reports wrong Ambiguous field identifier error when derived tables share column names

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#14372]
Using this schema:

```sql
create table t (i int, j int);
create table u (i int, j int);
```

And with `Settings.parseWithMetaLookups = THROW_ON_FAILURE`, the following query fails, even if it is perfectly fine:

```sql
select i
from (
  select x.i
  from (select i from t) as x, u
) y
```

The error is:

```
Ambiguous field identifier: [1:8] select [*]i
from (
  select x.i
  from (select i from t) as x, u
) y
```

Note that if the `u` table is removed from `y`:

```sql
select i
from (
  select x.i
  from (select i from t) as x
) y
```

Then a different field lookup error manifests:

```
Unknown field identifier: [1:8] select [*]i
from (
  select x.i
  from (select i from t) as x
) y
```

**Comment by lukaseder** [comment]
A related issue will be fixed first: https://github.com/jOOQ/jOOQ/issues/14373

**Comment by lukaseder** [comment]
The two problems are possibly related. The problem in the second query is that we're trying to resolve the outermost `i` column while resolving identifiers *within* the derived table `y`:

```sql
select i
from (
  select x.i
  from (select i from t) as x
-- Here, the outer most i is being resolved to x.i, which it would be if it were
-- located here. Alas, we shouldn't resolve it yet
) y

-- We should wait until we reach this point here until i is resolved to y.i
```

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.6 (https://github.com/jOOQ/jOOQ/issues/14376), and 3.16.12 (https://github.com/jOOQ/jOOQ/issues/14377)
